_posts/2016-06-23-Single-Responsibility-Principle.md

```diff
@@ -0,0 +1,108 @@
+---
+layout: post
+name: Single-Responsibility-Principle
+title: Some Thoughts on the Single Responsibility Principle
+date: 2016-06-22 10:00:00 +00:00
+author: Toby Retallick
+canonical:
+    name: my personal blog
+    href: http://tobyret.github.io/Single-Responsibility-Principle
+image:
+    src: /assets/img/custom/blog/2016-06-23-single-responsibility-principle.jpg
+tags:
+- SOLID
+- single responsibility
+- software design  
+---
+
+[SOLID](http://bit.ly/1Da1b16) (single responsibility, open-closed, Liskov substitution, interface segregation and dependency inversion) is a set of principles popularised by [‘Uncle’ Bob Martin](https://en.wikipedia.org/wiki/Robert_Cecil_Martin) that aim to guide developers in implementing good software design. The principles define how Object Oriented classes should relate to each other in order to create a codebase that is flexible and able to accommodate ongoing change.
+
+The first of these, the **Single Responsibility Principle** (SRP) states that -
+
+<blockquote class="hero">
+	<p><em>"Every module or class should have one, and only one reason to change."</em></p>
```

Comment:
This was wikipedia and so this definition appears to me to be the outside view of SRP.

What you wrote appears to be the inside view.

I'll review further and give you more feedback, as SRP casts its net quite wide into all the different principles.